### PR TITLE
feat: add box select

### PR DESCRIPTION
**Pull request by llaenowyd**
mishkaio_box_select_grid

mishkaio_box_select_1_1

Note that the original version of the above was first submitted by @mishkaio under [#57](https://github.com/featherity/featherity/pull/57).

### CODE REVIEW

**Comment by mishkaio** [comment]
@llaenowyd I'm not really clear on what I'm supposed to do now, if anything, or if you were just informing me. I have zero familiarity with this whole process, I was just trying to contribute as a designer. If still like to help, I just need clear directions.

**Comment by johnletey** [comment]
@llaenowyd Thank you for taking the time to update @mishkaio's icons.

I think it's best if you commented on the original PR, #57, and @mishkaio can update them herself.

Again, thank you for taking the time 😄

**Comment by ericfennis** [comment]
I reopened this issue, I don't get any response on #57 .
So @llaenowyd feel free to submit your versions.

**Comment by llaenowyd** [comment]
> I reopened this issue, I don't get any response on #57 .
> So @llaenowyd feel free to submit your versions.

Ok. I'm completely busy through the end of this week.

I took a look at the other PR. I think the latest feedback is to try something more of a dashed-line rectangle.

What I might do is focus the change down to the selection rectangle.

Is the version with cursor semantically different or a stylistic option?

Unless I'm missing something about it, I might bring in the box-with-cursor as an alternative possibility for box-select, rather than a request for 2 icons for box-select.

**Comment by ericfennis** [comment]
@llaenowyd No problem, take your time!

**Comment by llaenowyd** [comment]
The latest review feedback was when @ericfennis said

> Btw we can also go more this way? ...
> image

**Comment by llaenowyd** [comment]
dashed_line_square_grid
dashed_line_square_1_1

File: icons/box-select.svg [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <line x1="2" y1="2" x2="2" y2="4"/>
+  <line x1="2" y1="2" x2="4" y2="2"/>
```
Comment (ericfennis) [inline icons/box-select.svg:13]:
```suggestion
  <polyline points="2 4 2 2 4 2" />
```

File: icons/box-select.svg [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  <line x1="8" y1="2" x2="10" y2="2"/>
+  <line x1="14" y1="2" x2="16" y2="2"/>
+  <line x1="20" y1="2" x2="22" y2="2"/>
+  <line x1="22" y1="2" x2="22" y2="4"/>
```
Comment (ericfennis) [inline icons/box-select.svg:17]:
```suggestion
  <polyline points="20 2 22 2 22 4" />
```

File: icons/box-select.svg [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  <line x1="22" y1="8" x2="22" y2="10"/>
+  <line x1="22" y1="14" x2="22" y2="16"/>
+  <line x1="22" y1="20" x2="22" y2="22"/>
+  <line x1="20" y1="22" x2="22" y2="22"/>
```
Comment (ericfennis) [inline icons/box-select.svg:21]:
```suggestion
  <polyline points="22 20 22 22 20 22" />
```

File: icons/box-select.svg [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  <line x1="14" y1="22" x2="16" y2="22"/>
+  <line x1="8" y1="22" x2="10" y2="22"/>
+  <line x1="2" y1="22" x2="4" y2="22"/>
+  <line x1="2" y1="20" x2="2" y2="22"/>
```
Comment (ericfennis) [inline icons/box-select.svg:25]:
```suggestion
  <polyline points="2 20 2 22 4 22" />
```

**Review by ericfennis — CHANGES_REQUESTED** [review]
@llaenowyd Nice work, looks awesome.
I think we can optimize the code a bit by replacing some line elements with polylines.

File: icons/box-select.svg [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <polyline points="2 4 2 2 4 2" />
+  <line x1="8" y1="2" x2="10" y2="2"/>
+  <line x1="14" y1="2" x2="16" y2="2"/>
```
Comment (ericfennis) [inline icons/box-select.svg:14]:
```suggestion
  <line x1="8" y1="2" x2="10" y2="2" />
  <line x1="14" y1="2" x2="16" y2="2" />
```

File: icons/box-select.svg [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  <polyline points="20 2 22 2 22 4" />
+  <line x1="22" y1="8" x2="22" y2="10"/>
+  <line x1="22" y1="14" x2="22" y2="16"/>
```
Comment (ericfennis) [inline icons/box-select.svg:17]:
```suggestion
  <line x1="22" y1="8" x2="22" y2="10" />
  <line x1="22" y1="14" x2="22" y2="16" />
```

File: icons/box-select.svg [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  <polyline points="22 20 22 22 20 22" />
+  <line x1="14" y1="22" x2="16" y2="22"/>
+  <line x1="8" y1="22" x2="10" y2="22"/>
```
Comment (ericfennis) [inline icons/box-select.svg:20]:
```suggestion
  <line x1="14" y1="22" x2="16" y2="22" />
  <line x1="8" y1="22" x2="10" y2="22" />
```

File: icons/box-select.svg [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  <polyline points="2 20 2 22 4 22" />
+  <line x1="2" y1="14" x2="2" y2="16"/>
+  <line x1="2" y1="8" x2="2" y2="10"/>
```
Comment (ericfennis) [inline icons/box-select.svg:23]:
```suggestion
  <line x1="2" y1="14" x2="2" y2="16" />
  <line x1="2" y1="8" x2="2" y2="10" />
```

**Review by ericfennis — CHANGES_REQUESTED** [review]
@llaenowyd We always add an space before the end of the tag. Sorry I missed those.

**Comment by llaenowyd** [comment]
np - missed that - updated

**Comment by ericfennis** [comment]
I checked the guidelines, I think we forgot one thing. 
We have 2px rounded corners. 

What do you think? I think this looks better.  

image
100%
image

Here is the code
```svg
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M4 2a2 2 0 00-2 2" />
  <line x1="8" y1="2" x2="10" y2="2" />
  <line x1="14" y1="2" x2="16" y2="2" />
  <path d="M4 22a2 2 0 01-2-2" />
  <line x1="22" y1="8" x2="22" y2="10" />
  <line x1="22" y1="14" x2="22" y2="16" />
  <path d="M22 20a2 2 0 01-2 2" />
  <line x1="14" y1="22" x2="16" y2="22" />
  <line x1="8" y1="22" x2="10" y2="22" />
  <path d="M20 2a2 2 0 012 2" />
  <line x1="2" y1="14" x2="2" y2="16" />
  <line x1="2" y1="8" x2="2" y2="10" />
</svg>
```

**Comment by locness3** [comment]
Yup, rounded looks better.

**Comment by llaenowyd** [comment]
same, looks good!

**Review by ericfennis — APPROVED** [review]

**Comment by llaenowyd** [comment]
🎉 

I think I can consider this done for my part right?

What's the rough estimated timeline for it showing up in [npm feather-icons](https://www.npmjs.com/package/feather-icons)?

**Comment by ericfennis** [comment]
@llaenowyd unfortunately maybe never.
The author of feather-icons stopped maintaining the feather-icons projects 😔.
Featherity is just an independent fork runned by the community some of people worked on Feather icons.
See this [thread](https://github.com/feathericons/feather/issues/893), how it all started.

**Comment by ericfennis** [comment]
@llaenowyd But we are launching our own package soon!

**Comment by llaenowyd** [comment]
Ok, I see, interesting. I've seen dead packages and live packages, but I've never watched what looks to be a well motivated community effort to resuscitate a package. Hope to see it land. 👍

**Pull request merged**